models/src/main/scala/coop/rchain/models/rholang/implicits.scala

```diff
@@ -502,4 +502,131 @@ object implicits {
           case _ => BitSet()
         }
     }
+
+  implicit val ChannelIsForgeable: IsForgeable[Channel] =
+    new IsForgeable[Channel] {
+      def isForgeable(chan: Channel) =
+        chan.channelInstance match {
+          case Quote(p) => ParIsForgeable(p)
+          case _        => true
```

Comment:
```scala
case ChanVar(Var(BoundVar(…))) => 
```
if it binds to unforgeable name that this should be `false` right?

Reply:
what worries me is that there isn't a single `false` literal in the added code :)

Also I'd expect a case for a GPrivate.

Reply:
Because it is `IsForgeable` - it checks whether the `Par` is forgeable. Being not forgeable is just the negation.

Reply:
There's no `GPrivate` explicitly but the check is here:
> par.ids.isEmpty

Reply:
I'd say this case should be done using forAll, and the GPrivate case should be explicit. Ohterwise we're making a strong assumption that a GPrivate can only appear in par.ids. Which could change in the future, and I wouldn't like to be the miser debugging this.

Reply:
I disagree. We explicitly separated GPrivates to make this easy. I can add the instance for completeness if it bugs you, but it's unused code.